SPECS/postgresql/postgresql.spec

```diff
@@ -101,10 +105,59 @@ chown -Rv nobody .
 sudo -u nobody -s /bin/bash -c "PATH=$PATH make -k check"
 
 %ldconfig_scriptlets
+%pre
+
+if ! getent group postgres >/dev/null; then
+    /sbin/groupadd -r postgres
+fi
+
+if ! getent passwd postgres >/dev/null; then
+    /sbin/useradd -g postgres postgres
+fi
+
+%post
+
+PGDATA="/usr/local/pgsql/data"
+PGRUN="/run/postgresql"
+
+if [ ! -d "$PGDATA" ]; then
+    mkdir -p "$PGDATA"
+    chown postgres:postgres "$PGDATA"
+    sudo -u postgres /usr/bin/initdb -D "$PGDATA"
+fi
+
+chown -R postgres:postgres "$PGDATA"
```

Comment:
As this is done on an installation of a new package, should this be put into the above **if** block? My reasoning is, what if someone already has a mount there with a different UID, could inadvertently change the uid without the user realising.